src/test/integrationTests/userController.test.js

```diff
@@ -143,4 +143,137 @@ describe('Integration tests for the user controller', () => {
+        expect(result.body.data.userDetails.lineManager).to
+          .equal(newData.lineManager);
+      });
+
+      it('should return error when profile is updated with invalid data', async () => {
+        const result = await chai.request(app).patch(`/api/v1/profile/${id}`)
+          .set('x-access-token', token)
+          .send({
+            firstName: 'rt',
+            lastName: 'hi',
+            birthdate: 'ehegw',
+            gender: 'dj',
+          });
+        expect(result.body).to.have.property('success');
+        expect(result.body.success).to.equal(false);
+        expect(result.body).to.have.property('message');
+        expect(result.body.message).to.be.an('object');
+        expect(result.body.message.name).to.equal('SequelizeValidationError');
+      });
+
+      it('should return error when profile is updated with invalid token', async () => {
+        const result = await chai.request(app).patch(`/api/v1/profile/${id}`)
+          .set('x-access-token', 'this is an invalid token')
+          .send(newData);
+        expect(result.body.success).to.equal(false);
+        expect(result.body.message).equal('User not authorized');
+      });
+
+      it('should return error when profile is updated by another user', async () => {
+        const anotherUsersId = 'this_id_will not work';
+        const result = await chai.request(app).patch(`/api/v1/profile/${anotherUsersId}`)
+          .set('x-access-token', token)
+          .send(newData);
+        expect(result.body.success).to.equal(false);
+        expect(result.body.message).to
+          .equal('you are not authorized to perform this action');
+      });
+
+      it('should fetch users current profile settings', async () => {
+        const result = await chai.request(app).get(`/api/v1/profile/${id}`)
+          .set('x-access-token', token);
+        expect(result.body.data.success).to.equal(true);
+        expect(result.body.data.userDetails.firstName).to
+          .equal(newData.firstName);
+        expect(result.body.data.userDetails.lastName).to
+          .equal(newData.lastName);
+        expect(result.body.data.userDetails.preferredLanguage).to
+          .equal(newData.preferredLanguage);
+        expect(result.body.data.userDetails.preferredCurrency).to
+          .equal(newData.preferredCurrency);
+        expect(result.body.data.userDetails.city).to
+          .equal(newData.city);
+        expect(result.body.data.userDetails.state).to
+          .equal(newData.state);
+        expect(result.body.data.userDetails.zip).to
+          .equal(newData.zip);
+        expect(result.body.data.userDetails.country).to
+          .equal(newData.country);
+        expect(result.body.data.userDetails.role).to
+          .equal(newData.role);
+        expect(result.body.data.userDetails.department).to
+          .equal(newData.department);
+        expect(result.body.data.userDetails.lineManager).to
+          .equal(newData.lineManager);
+      });
```

Comment:
Remember not to send the user's password. There is a method that should make you select only specific fields/columns.